conda_smithy/configure_feedstock.py

```diff
@@ -522,7 +605,7 @@ def _collapse_subpackage_variants(
     top_level_loop_vars = set()
 
     all_used_vars = set()
-    all_variants = set()
+    all_variants: Any = set()
 
```

Comment:
top_level_loop_vars and all_used_vars are already set?

Reply:
for some reason mypy was complaining that `all_variants` needs a type. So I added `Any` but now I updated it and added type as `set`. Also updated the function `_merge_deployment_target` on line 506 to `set` for the types to align.